### PR TITLE
K8SPXC-519: operator logs improvements

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPXC-519](https://jira.percona.com/browse/K8SPXC-519) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

Add cluster name and namespace to log messages
Refactor errors

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| storage | passed |
| haproxy | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | failed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| security-context | passed |
| limits | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| demand-backup | passed |
| affinity | passed |
| self-healing-advanced | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| users | passed |
| pitr | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-725-07b7d162

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| security-context | passed |
| self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| haproxy | passed |
| demand-backup | passed |
| init-deploy | passed |
| pitr | passed |
| limits | passed |
| demand-backup-encrypted-with-tls | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| upgrade-haproxy | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |

File: pkg/controller/pxc/controller.go [hunk 1]
```diff
@@ -199,35 +195,36 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	reqLogger := log.WithValues("cluster name", o.Name, "namespace", o.Namespace)
```
Comment (blez) [inline pkg/controller/pxc/controller.go:198]: I think It's better to have global logger and not to pass it as a argument.

Comment (qJkee) [reply pkg/controller/pxc/controller.go:198]: we can't have global logger because requests could come in parallel

Comment (blez) [reply pkg/controller/pxc/controller.go:198]: Maybe we can have logger store then?

Comment (qJkee) [reply pkg/controller/pxc/controller.go]: 6af1e46

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| haproxy | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| affinity | passed |
| smart-update | passed |
| demand-backup | passed |
| one-pod | passed |
| auto-tuning | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | failed |
| pitr | passed |
| users | failed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | failed |

**Review by delgod — DISMISSED** [review]

File: pkg/controller/pxc/version.go [hunk 2]
```diff
@@ -267,31 +271,33 @@ func (r *ReconcilePerconaXtraDBCluster) fetchVersionFromPXC(cr *api.PerconaXtraD
...
 
 		version, err := database.Version()
 		if err != nil {
-			log.Error(err, "failed to get pxc version")
+			logger.Error(err, "failed to get pxc version")
 			continue
 		}
 
-		log.Info(fmt.Sprintf("update PXC version to %v (fetched from db)", version))
+		logger.Info(fmt.Sprintf("update PXC version to %v (fetched from db)", version))
```
Comment (blez) [inline pkg/controller/pxc/version.go:291]: Maybe you forgot to get rid of `fmt` here

**Review by blez — DISMISSED** [review]

**Review by hors — DISMISSED** [review]

File: pkg/controller/pxc/version.go (hunk 2 above)
Comment (qJkee) [reply pkg/controller/pxc/version.go]: yea)
16aa8e77

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| haproxy | passed |
| big-data | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| scaling-proxysql | passed |
| limits | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| smart-update | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| self-healing-advanced | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | failed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

**Pull request merged**